### ISSUE TITLE
[CI] FullClusterRestartIT.testSearch and CoreFullClusterRestartIT.testSearch fail

### ISSUE THREAD

**Issue by imotov** [elastic/elasticsearch#63366]
**Build scan**: https://gradle-enterprise.elastic.co/s/6gb4olqyopyea

**Repro line**:
```
REPRODUCE WITH: ./gradlew ':qa:full-cluster-restart:v6.2.1#upgradedClusterTest' --tests "org.elasticsearch.upgrades.FullClusterRestartIT.testSearch" \
  -Dtests.seed=74D47C9642E75C9F \
  -Dtests.security.manager=true \
  -Dtests.locale=es-UY \
  -Dtests.timezone=Chile/EasterIsland \
  -Dtests.distribution=default \
  -Druntime.java=8

REPRODUCE WITH: ./gradlew ':x-pack:qa:full-cluster-restart:v6.2.1#upgradedClusterTest' --tests "org.elasticsearch.xpack.restart.CoreFullClusterRestartIT.testSearch" \
  -Dtests.seed=74D47C9642E75C9F \
  -Dtests.security.manager=true \
  -Dtests.locale=es-CU \
  -Dtests.timezone=America/Cuiaba \
  -Dtests.distribution=default \
  -Druntime.java=8
```

**Reproduces locally?**:

**Applicable branches**: 7.x

**Failure history**:

It fails approximately once a day in bursts for each 6.x.y version. It looks like it was introduced in 7405af8060bd32bf2ae7917a13c441e4214339b8 (I can only reproduce it starting with this commit) 

**Failure excerpt**:
```
org.elasticsearch.client.WarningFailureException: method [GET], host [http://[::1]:39853], URI [/testsearch/doc/571/_explain], status line [HTTP/1.1 200 OK]
Warnings: [[types removal] Specifying a type in explain requests is deprecated., [types removal] Using the _type field in queries and aggregations is deprecated, prefer to use a field instead.]
{"_index":"testsearch","_type":"doc","_id":"571","matched":true,"explanation":{"value":1.0,"description":"*:*","details":[]}}
	at __randomizedtesting.SeedInfo.seed([74D47C9642E75C9F:E66DE9C306FE0430]:0)
	at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:314)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:288)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:262)
	at org.elasticsearch.upgrades.FullClusterRestartIT.assertAllSearchWorks(FullClusterRestartIT.java:566)
	at org.elasticsearch.upgrades.FullClusterRestartIT.testSearch(FullClusterRestartIT.java:171)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by dakrone** [comment]
I believe this was just fixed by @gwbrown in #63363

**Comment by imotov** [comment]
That was fixed in #63391 but github didn't autoclose it because it was merged only into 7.x. Closing.
